File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let ans = 0;
    for (let i = 0; i < babbling.length; i++) {
        let flag = true;
        const checkItem = babbling[i];
        for (const item of ['aya', 'ye', 'woo', 'ma']) {
            // 두번 이상 반복한 것이 포함되면 같은 발음을 연속 발음한 것이므로
            // 반복을 멈추고 넘어간다.
            if (checkItem.includes(item.repeat(2))) {
                flag = false;
                break;
            }
        }
        // 두번 연속으로 발음하지 않았으면
        if (flag) {
            // 발음할 수 있는 단어인지 확인
            if (checkItem.replace(/aya|ye|woo|ma/g, '') === '') {
                ans++;
            }
        }
    }
    return ans;
}

File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma"]
    
    for item in babbling:
        flag = False
        for index, word in enumerate(words):
            item = item.replace(word, str(index))
            if str(index) + str(index) in item:
                flag = True
                break
        
        if not flag:
            if item.isdigit():
                answer += 1
            
    return answer